### ISSUE TITLE
When using `ConvertUsing` and it return json, `;` (delimiter) is missing on value lines.

### ISSUE THREAD

**Issue by latop2604** [JoshClose/CsvHelper#1595]
**Describe the bug**
When using `ConvertUsing` and it return json, `;` (delimiter) is missing on value lines.

**To Reproduce**
```csharp
async Task Main()
{
	await T(new LineMapKO());
	await T(new LineMapOK());
}
async Task T(ClassMap classMap)
{
	var lines = new List<Line>();
	lines.Add(new Line{
		Id = 1,
		Name = "toto",
		Date = DateTime.UtcNow,
		Date2 = null,
		Comment = "No!",
	});
	
	var ms = new MemoryStream();
	
	var sw = new StreamWriter(ms, new UTF8Encoding(false), leaveOpen: true);
	var writer = new CsvHelper.CsvWriter(sw, CultureInfo.InvariantCulture);
	writer.Configuration.Delimiter = ";";
	writer.Configuration.RegisterClassMap(classMap);

	await writer.WriteRecordsAsync(lines);
	await writer.FlushAsync();
	await sw.FlushAsync();
	
	ms.Position = 0;
	var sr = new StreamReader(ms, new UTF8Encoding(false));
	var result = await sr.ReadToEndAsync();
	Console.WriteLine(result);
}

public class Line
{
	public int Id { get; set; }
	public string Name { get; set; }
	public DateTime Date { get; set; }
	public DateTime? Date2 { get; set; }
	public string Comment { get; set; }
}

public class LineMapKO : ClassMap<Line>
{
	public LineMapKO()
	{
		Map(l => l.Id);
		Map(l => l.Name);
		Map(l => l.Date).ConvertUsing(l => l.Date.ToString("O", CultureInfo.InvariantCulture));
		Map(l => l.Date2).ConvertUsing(l => l.Date2?.ToString("O", CultureInfo.InvariantCulture));
		Map(l => l.Comment);
	}
}
public class LineMapOK : ClassMap<Line>
{
	public LineMapOK()
	{
		Map(l => l.Id);
		Map(l => l.Name);
		Map(l => l.Date).ConvertUsing(l => l.Date.ToString("O", CultureInfo.InvariantCulture));
		Map(l => l.Date2).ConvertUsing(l => l.Date2?.ToString("O", CultureInfo.InvariantCulture)?? string.Empty);
		Map(l => l.Comment);
	}
}

```

**Expected behavior**
```csv
Id;Name;Date;Date2;Comment
1;toto;2020-10-12T14:59:54.1169637Z;;No!
```

**Comment by JoshClose** [comment]
Weird... I'll look into this.

**Comment by JoshClose** [comment]
This is fixed an in release 15.0.7 on nuget.
